### PR TITLE
[ZH DateTimeV2] Date regression fix due to side-effect of previous Range change (#2375)

### PR DESCRIPTION
**Pull request by ChopperMan33**
Refer to discussion #2375, recognizing a purely numeric date in Chinese will follow these rules:
1. If the order/format is X/X/Y, treat it as D/M/Y. 
2. If the order/format is Y/X/X, treat it as Y/M/D.
3. If the order/format is X/X, treat it as M/D.

Added test cases.

### CODE REVIEW

File: Patterns/Chinese/Chinese-DateTime.yaml [hunk 1]
```diff
@@ -85,24 +85,23 @@ DateRegexList3: !nestedRegex
...
+  def: '{MonthNumRegex}\s*/\s*{DayRegex}'
+  references: [MonthNumRegex, DayRegex]
 # 23/7
 DateRegexList5: !nestedRegex
-  def: '{DayRegex}\s*/\s*{MonthNumRegex}((\s+|\s*,\s*){SimpleYearRegex})?'
-  references: [DayRegex, MonthNumRegex, SimpleYearRegex]
+  def: '{DayRegex}\s*/\s*{MonthNumRegex}'
+  references: [DayRegex, MonthNumRegex]
 # 3-23-2017
-# The final lookahead in DateExtractor6|7|8 avoids extracting as date "10/1-11" from an input like "10/1-11/2/2017"
```
Comment (ChopperMan33) [inline Patterns/Chinese/Chinese-DateTime.yaml:95]: "10/1-11/2/2017" is not a format used. This code used to implement it has caused cases like "10/1/2018-10/7/2018" to be incorrectly extracted.

File: .NET/Microsoft.Recognizers.Text.DateTime/Constants.cs [hunk 2]
```diff
@@ -121,6 +121,7 @@ public static class Constants
 
         public const string DefaultLanguageFallback_MDY = "MDY";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Constants.cs:122]: The comment is missing here.

File: .NET/Microsoft.Recognizers.Definitions.Common/Chinese/DateTimeDefinitions.cs [hunk 3]
```diff
@@ -55,11 +55,11 @@ public static class DateTimeDefinitions
...
-      public static readonly string DateRegexList4 = $@"{MonthNumRegex}\s*/\s*{DayRegex}((\s+|\s*,\s*){SimpleYearRegex})?";
-      public static readonly string DateRegexList5 = $@"{DayRegex}\s*/\s*{MonthNumRegex}((\s+|\s*,\s*){SimpleYearRegex})?";
-      public static readonly string DateRegexList6 = $@"{MonthNumRegex}\s*[/\\\-]\s*{DayRegex}\s*[/\\\-]\s*{SimpleYearRegex}(?!\s*[/\\\-\.]\s*\d+)";
-      public static readonly string DateRegexList7 = $@"{DayRegex}\s*[/\\\-\.]\s*{MonthNumRegex}\s*[/\\\-\.]\s*{YearNumRegex}(?!\s*[/\\\-\.]\s*\d+)";
-      public static readonly string DateRegexList8 = $@"{YearNumRegex}\s*[/\\\-\. ]\s*{MonthNumRegex}\s*[/\\\-\. ]\s*{DayRegex}(?!\s*[/\\\-\.]\s*\d+)";
+      public static readonly string DateRegexList4 = $@"{MonthNumRegex}\s*/\s*{DayRegex}";
+      public static readonly string DateRegexList5 = $@"{DayRegex}\s*/\s*{MonthNumRegex}";
+      public static readonly string DateRegexList6 = $@"{MonthNumRegex}\s*[/\\\-]\s*{DayRegex}\s*[/\\\-]\s*{SimpleYearRegex}";
+      public static readonly string DateRegexList7 = $@"{DayRegex}\s*[/\\\-\.]\s*{MonthNumRegex}\s*[/\\\-\.]\s*{SimpleYearRegex}";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions.Common/Chinese/DateTimeDefinitions.cs:61]: Why this change from YearNumRegex?

File: .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Extractors/ChineseDateExtractorConfiguration.cs [hunk 4]
```diff
@@ -75,38 +75,25 @@ public class ChineseDateExtractorConfiguration : AbstractYearExtractor, IDateTim
             // (2015年)?(农历)?十月二十(星期三)?
             new Regex(DateTimeDefinitions.DateRegexList3, RegexFlags),
 
-            DateTimeDefinitions.DefaultLanguageFallback == Constants.DefaultLanguageFallback_DMY ?
+            // 7/23
+            new Regex(DateTimeDefinitions.DateRegexList4, RegexFlags),
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Extractors/ChineseDateExtractorConfiguration.cs:79]: Why is this no longer under the flag check?

File: .NET/Microsoft.Recognizers.Definitions.Common/Chinese/DateTimeDefinitions.cs [hunk 5]
```diff
@@ -55,11 +55,11 @@ public static class DateTimeDefinitions
       public static readonly string DateRegexList1 = $@"({LunarRegex}(\s*))?((({SimpleYearRegex}|{DateYearInChineseRegex})年)(\s*))?{MonthRegex}(\s*){DateDayRegexInChinese}((\s*|,|，){WeekDayRegex})?";
       public static readonly string DateRegexList2 = $@"((({SimpleYearRegex}|{DateYearInChineseRegex})年)(\s*))?({LunarRegex}(\s*))?{MonthRegex}(\s*){DateDayRegexInChinese}((\s*|,|，){WeekDayRegex})?";
       public static readonly string DateRegexList3 = $@"((({SimpleYearRegex}|{DateYearInChineseRegex})年)(\s*))?({LunarRegex}(\s*))?{MonthRegex}(\s*)({DayRegexNumInChinese}|{DayRegex})((\s*|,|，){WeekDayRegex})?";
-      public static readonly string DateRegexList4 = $@"{MonthNumRegex}\s*/\s*{DayRegex}((\s+|\s*,\s*){SimpleYearRegex})?";
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions.Common/Chinese/DateTimeDefinitions.cs:58]: Had that commit that caused the side-effect also changed these? I thought it only changed regexes 5, 6, and 7. But maybe I misremember.

File: .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Extractors/ChineseDateExtractorConfiguration.cs [hunk 6]
```diff
@@ -75,38 +75,25 @@ public class ChineseDateExtractorConfiguration : AbstractYearExtractor, IDateTim
...
-                // 23/7
-                new Regex(DateTimeDefinitions.DateRegexList5, RegexFlags),
+                // 23-3-2017
+                new Regex(DateTimeDefinitions.DateRegexList7, RegexFlags),
 
-            DateTimeDefinitions.DefaultLanguageFallback == Constants.DefaultLanguageFallback_DMY ?
+            DateTimeDefinitions.DefaultLanguageFallback == Constants.DefaultLanguageFallback_MDY ?
 
-                // 23-3-2015
+                // 23-3-2017
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Extractors/ChineseDateExtractorConfiguration.cs:91]: No need to change the year. :)

File: Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/Constants.java [hunk 7]
```diff
@@ -116,6 +116,7 @@
 
     public static final String DefaultLanguageFallback_MDY = "MDY";
```
Comment (tellarin) [inline Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/Constants.java:117]: Same.

File: JavaScript/packages/recognizers-date-time/src/dateTime/constants.ts [hunk 8]
```diff
@@ -35,6 +35,7 @@ export class Constants {
 
     static readonly DefaultLanguageFallback_MDY: string = 'MDY';
```
Comment (tellarin) [inline JavaScript/packages/recognizers-date-time/src/dateTime/constants.ts:36]: Same

File: Patterns/Chinese/Chinese-DateTime.yaml (hunk 1 above)
Comment (tellarin) [reply Patterns/Chinese/Chinese-DateTime.yaml:95]: Does it now extract cases like "10/1/2018-10/7/2018"  and "2018/1/10-2018/7/10" correctly?

**Review by tellarin — CHANGES_REQUESTED** [review]

File: .NET/Microsoft.Recognizers.Definitions.Common/Chinese/DateTimeDefinitions.cs (hunk 3 above)
Comment (ChopperMan33) [reply .NET/Microsoft.Recognizers.Definitions.Common/Chinese/DateTimeDefinitions.cs:61]: The YearNumRegex can only support years that contain 4 numbers like 1999 2020. Now SimpleYearRegex can support years that only contain 2 numbers like "3-7-07" -> ”2007-07-03“, which is same use in DateRegexList6.

File: .NET/Microsoft.Recognizers.Definitions.Common/Chinese/DateTimeDefinitions.cs (hunk 5 above)
Comment (ChopperMan33) [reply .NET/Microsoft.Recognizers.Definitions.Common/Chinese/DateTimeDefinitions.cs:58]: Because the order of month and day in X/X(M/D) is different from X/X/Y(D/M/Y), I changed DateRegexList4/5 to only support X/X pattern. Separately extract X/X and X/X/Y pattern.

File: .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Extractors/ChineseDateExtractorConfiguration.cs (hunk 4 above)
Comment (ChopperMan33) [reply .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Extractors/ChineseDateExtractorConfiguration.cs:79]: For now, DataRegexList4 only extract "X/X" pattern,  in Chinese, only use M/D in "X/X" pattern according to the rules.

File: .NET/Microsoft.Recognizers.Text.DateTime/Constants.cs (hunk 2 above)
Comment (ChopperMan33) [reply .NET/Microsoft.Recognizers.Text.DateTime/Constants.cs:122]: Sorry for that, I will add it immediately.

File: Patterns/Chinese/Chinese-DateTime.yaml (hunk 1 above)
Comment (ChopperMan33) [reply Patterns/Chinese/Chinese-DateTime.yaml:95]: Yes. I will add them into specs.

File: Specs/DateTime/Chinese/DateParser.json [hunk 9]
```diff
@@ -909,12 +909,12 @@
         "Text": "3-7-27",
```
Comment (tellarin) [inline Specs/DateTime/Chinese/DateParser.json:909]: This one is interesting though. In the two digit year case, would 27-7-3 return 2027-07-03 too? It should as YMD should have precedence over DMY.

**Review by tellarin — CHANGES_REQUESTED** [review]

**Review by tellarin — APPROVED** [review]
